packages/compass-components/src/components/links/link.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, forwardRef } from 'react';
+import { Link as LGLink } from '@leafygreen-ui/typography';
+import LGButton from '@leafygreen-ui/button';
+import LGIconButton from '@leafygreen-ui/icon-button';
+
+type LinkContextValue = {
+  utmSource?: string;
+  utmMedium?: string;
+};
+const LinkContext = React.createContext<LinkContextValue>({});
+export const LinkProvider: React.FC<LinkContextValue> = ({
+  utmSource,
+  utmMedium,
+  children,
+}) => {
+  const value: LinkContextValue = {
+    utmSource,
+    utmMedium,
+  };
+
+  return <LinkContext.Provider value={value}>{children}</LinkContext.Provider>;
+};
+
+// exported for testing purposes only
+export const EXCLUDED_MONGODB_HOSTS = [
+  'compass-maps.mongodb.com',
+  'evergreen.mongodb.com',
+  'downloads.mongodb.com',
+  'cloud.mongodb.com',
+];
+
+export const urlWithUtmParams = (
+  urlString: string,
+  { utmSource, utmMedium }: { utmSource?: string; utmMedium?: string }
+): string => {
+  try {
+    const url = new URL(urlString);
+    const urlShouldHaveUtmParams =
+      /^(.*\.)?mongodb\.com$/.test(url.hostname) &&
+      !EXCLUDED_MONGODB_HOSTS.includes(url.hostname);
+
+    if (urlShouldHaveUtmParams) {
+      if (utmSource && !url.searchParams.has('utm_source')) {
+        url.searchParams.set('utm_source', utmSource);
+      }
+      if (utmMedium && !url.searchParams.has('utm_medium')) {
+        url.searchParams.set('utm_medium', utmMedium);
+      }
+    }
+    return url.toString();
+  } catch {
+    return urlString;
+  }
+};
+
+export function useLinkContext(): LinkContextValue {
+  return useContext(LinkContext);
+}
+
+export const Link = (({
+  href,
+  children,
+  ...rest
+}: React.ComponentProps<typeof LGLink>) => {
+  const { utmSource, utmMedium } = useLinkContext();
+
+  if (href) {
+    href = urlWithUtmParams(href, { utmSource, utmMedium });
```

Comment:
Should probably be a memo so that we don't need to parse, construct, and stringify url every time any link on the screen is re-rendered